echoprometheus/prometheus.go

```diff
@@ -267,7 +271,10 @@ func (conf MiddlewareConfig) ToMiddleware() (echo.MiddlewareFunc, error) {
 			values[0] = strconv.Itoa(status)
 			values[1] = c.Request().Method
 			values[2] = c.Request().Host
-			values[3] = url
+			// as of Echo v4.10.1 an empty c.Path() means the router did not find any matching routes (404)
+			if c.Path() != "" || !conf.DoNotUseURLFor404 {
```

Comment:
This condition should be at line 249 `if url == "" {` so it would look like `if url == "" && !conf.DoNotUseURLFor404 {`  In that case we have that logic in one place and not divided between 2 separate places/blocks, which makes mentally harder to understand what the middleware does.

and considering `url = c.Request().URL.Path` the flag name could be `DoNotFillURLFor404` or  `DoNotUseRequestPathFor404`

Reply:
Thanks a lot! Fixed it.